ql/src/test/results/clientnegative/avro_non_nullable_union.q.out

```diff
@@ -34,5 +34,39 @@ PREHOOK: query: INSERT OVERWRITE TABLE union_non_nullable_test_avro SELECT * FRO
 PREHOOK: type: QUERY
 PREHOOK: Input: default@union_nullable_test_text
 PREHOOK: Output: default@union_non_nullable_test_avro
+Status: Failed
+Vertex failed, vertexName=Map 1, vertexId=vertex_#ID#, diagnostics=[Task failed, taskId=task_#ID#, diagnostics=[TaskAttempt 0 failed, info=[Error: Error while running task ( failure ) : attempt_#ID#:java.lang.RuntimeException: java.lang.RuntimeException: Hive Runtime Error while closing operators
 #### A masked pattern was here ####
-FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask
+Caused by: java.lang.RuntimeException: Hive Runtime Error while closing operators
+#### A masked pattern was here ####
+Caused by: org.apache.hadoop.hive.ql.metadata.HiveException: Avro could not validate record against schema (record = {"id": 3, "value": null}) (schema = {"type":"record","name":"nullable","fields":[{"name":"id","type":"int"},{"name":"value","type":["int","double"]}]})
+#### A masked pattern was here ####
+Caused by: Avro could not validate record against schema (record = {"id": 3, "value": null}) (schema = {"type":"record","name":"nullable","fields":[{"name":"id","type":"int"},{"name":"value","type":["int","double"]}]})
```

Comment:
It is not exactly that. The same stack trace is seen twice because tez.am.task.max.failed.attempts is set to 2 in tez-site.xml. If you look carefully you can see that first exception is because `TaskAttempt 0` failed and the second one is because `TaskAttempt 1` failed. This is the case for all the tests.

I thought about setting the config to 1 but I did not beause:
1. The config file is shared with positive tests and surely there are tests depend on second attempt on both positive and negative tests side. This config is not configurable in runtime. So we will probably need to introduce another driver to separate tests that require two attempts and the ones that are ok with one attempt.
2. Once you know Tez attempts tasks twice, the output does not look so messy to me. It shows exactly what happened. Also, attempt count is almost never set to 1 in production. So I think this is a better test than single-attempt test.
3. We can catch if the two attempts fail for different reasons somehow. Like in a scenario where first attempt fails with the expected exception but the second attempt goes rogue because we mess with something mutable in the first run.

On the other hand, setting attempt count to 1 will reduce test time for negative tests by roughly 50%.

I am leaning on keeping the attempt count at 2 and the output as verbose as it is right now. What do you think?

Reply:
Actually about reducing run-time, it is not going to make that much difference overall because majority of negative tests fail in Hiveserver without ever starting a tez job.